.travis.yml

```diff
@@ -13,12 +10,15 @@ python:
 - 3.6
 - &mainstream_python 3.7
 - nightly
-- &pypy3 pypy3.5-6.0.0
+- pypy3.5-6.0.0
 
 install:
 - &upgrade_python_toolset pip install --upgrade pip wheel setuptools
+- pip install --upgrade -r requirements/cython.txt
```

Comment:
Why do you apply `--upgrade` here and below?
Our requirement files always have strict dependency versions, updated by dependabot.
I thought that `-U` is pointless in this case.

Reply:
I've used `--upgrade` throughout to be consistent, and to account for possible 'comes with Cython pre-installed' issues. It may be better to default to *not* using `--upgrade` and explicitly documenting when we do need to use the flag (e.g. for `pip` itself).

Reply:
> It may be better to default to not using --upgrade and explicitly documenting when we do need to use the flag (e.g. for pip itself).

You've got my point. In the project we are *implicitly* follow this style (at least I think we do). If you want to document this fact somewhere -- you are welcome :)